docs: Update Ollama guide documentation

bdougie opened the pull request:
## Description

Updated the Ollama guide documentation with comprehensive improvements including:
- Added detailed setup instructions for macOS, Windows, and Linux
- Included configuration examples for different model types (chat, autocomplete, embeddings)
- Added troubleshooting section with common issues and solutions
- Improved formatting and organization for better readability
- Added examples of using specific models like Llama 3.1, CodeGemma, and Starcoder2

https://continue-docs-bdougie-con-3239.mintlify.app/guides/ollama-guide

## AI Code Review

- **Team members only**: AI review runs automatically when PR is opened or marked ready for review
- Team members can also trigger a review by commenting `@continue-general-review` or `@continue-detailed-review`

## Checklist

- [x] I've read the [contributing guide](https://github.com/continuedev/continue/blob/main/CONTRIBUTING.md)
- [x] The relevant docs, if any, have been updated or created
- [x] The relevant tests, if any, have been updated or created

## Screen recording or screenshot

N/A - Documentation update only

<img width="797" height="627" alt="Screenshot 2025-08-18 at 4 06 35 PM" src="https://github.com/user-attachments/assets/644834c8-bb91-43d8-817a-7927fa49c415" />
<img width="797" height="627" alt="Screenshot 2025-08-18 at 4 06 35 PM" src="https://github.com/user-attachments/assets/6eda769f-7401-4241-ac94-2541c0a1436f" />

## Tests

N/A - This is a documentation-only change that doesn't require tests

---

This PR updates the Ollama guide documentation with improvements and clarifications.

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
A few nitpicks on this guide that aren't part of the changes:

```diff
- Operating System: macOS, Linux, or Window
```

Patrick-Erichsen replied:
Maybe is worth pointing people towards the native desktop app, I use it: https://ollama.com/download

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
```diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
-ollama pull qwen2.5-coder:1.5b-base
-ollama pull nomic-embed-text:latest
+
+### Step 3: How to Download Models
```
Might be helpful to point to the directory for models: https://ollama.com/search

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
```diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
-### How to Set Up Continue Configuration
+### Method 1: Using Hub Model Blocks in Local config.yaml
+
+The easiest way is to use pre-configured model blocks from the Continue Hub in your local configuration:
```
Pointing to the ollama org on the Hub would be helpful here: https://hub.continue.dev/ollama

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
````diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+The easiest way is to use pre-configured model blocks from the Continue Hub in your local configuration:
 
 ```yaml
+# ~/.continue/assistants/My Local Assistant.yaml
````
```suggestion
# ~/.continue/assistants/assistant.yaml
```

Wondering if users won't know what this syntax is/what `.continue/assistants` is more generally. Might be useful to link to https://docs.continue.dev/reference#local-blocks

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
````diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+  - uses: ollama/gpt-oss-20b
+```
+
+<Warning>
+**Important**: Hub blocks only provide configuration - you still need to pull the model locally. The hub block `ollama/deepseek-r1-32b` configures Continue to use `model: deepseek-r1:32b`, but the actual model must be installed:
+```bash
+# Check what the hub block expects (view on hub.continue.dev)
+# Then pull that exact model tag locally
+ollama pull deepseek-r1:32b  # Required for ollama/deepseek-r1-32b hub block
+```
+If the model isn't installed, Ollama will return: `404 model "deepseek-r1:32b" not found, try pulling it first`
+</Warning>
````
This is nice 👌  definitely a common point of confusion

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
````diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+Continue can automatically detect available Ollama models. You can configure this in your YAML:
+
+```yaml
+models:
````
Should this also have a comment indicating that you should place it in `./continue/models/ollama-autodetect.yaml`?

For both this and the previous comment doing this via a title in the codefence might be nice UX.

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
```diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+
+Or use it through the GUI:
+
+1. Click on the model selector dropdown
```
The UX in the extension is kind of bad here imo. A few steps that are worth clarifying:
- You need to click "Add chat model" in the dropdown
  - But, you must be using a local assistant for that button to render
- Then select `Ollama` as the provider
<img width="353" height="537" alt="Screenshot 2025-08-20 at 2 51 09 PM" src="https://github.com/user-attachments/assets/ef90243c-a3f6-4a05-8fdb-04dc96515f46" />

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
```diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+The Autodetect feature scans your local Ollama installation and lists all available models. When set to `AUTODETECT`, Continue will dynamically populate the model list based on what's installed locally via `ollama list`. This is useful for quickly switching between models without manual configuration.
+</Note>
+
+You can update `apiBase` with the IP address of a remote machine serving Ollama.
```
This probably belongs in a `<Note />`

Patrick-Erichsen replied:
Hm, actually we don't want back to back notes. Maybe move it into a standalone and/or later section .

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
````diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+
+For custom configurations or models not on the hub:
+
+```yaml
````
Maybe add a title in the codefence for the local model yaml path

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
```diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
+
+Some Ollama models support tools (function calling) which is required for Agent mode. However, not all models that claim tool support work correctly:
+
+#### Checking Tool Support
```
Nice, very common problem

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
```diff
@@ -29,57 +29,162 @@ curl -fsSL https://ollama.ai/install.sh | sh
-- GPU acceleration
-- Custom model parameters
-- Performance tuning
+- Memory optimization: Adjust `num_ctx` for context window size
```
Where do you set these? I'm not sure myself tbh

Patrick-Erichsen commented on docs/guides/ollama-guide.mdx:
````diff
@@ -102,30 +207,68 @@ To get the best performance from Ollama:
+ollama pull deepseek-r1:32b
+```
+
+#### "Agent mode is not supported"
+
+**Problem**: Model doesn't support tools/function calling
+**Solutions**:
+1. Add `capabilities: [tool_use]` to your model config
+2. If still not working, the model may not actually support tools
+3. Switch to a model with confirmed tool support (Llama 3.1, Mistral)
````
I think this is a bit repetitive with sections above but probably nbd

Patrick-Erichsen reviewed (approved):
Bunch of nitpicks, lot of improvements here! Good to merge as-is though so feel free to not address all the comments.

bdougie commented:
Hey @Patrick-Erichsen! I've addressed all the review feedback:

✅ **Simplified section headers** - Removed redundant "How to" from subsections
✅ **Added version verification** - Included `ollama --version` command after installation  
✅ **Expanded model recommendations** - Added specific model names with memory requirements for different use cases
✅ **Added concrete advanced settings** - Provided YAML configuration examples with detailed parameters and Modelfile examples
✅ **Included diagnostic commands** - Added `ollama ps` and `ollama logs` for troubleshooting
✅ **Fixed code formatting** - Properly formatted the FastAPI Python example
✅ **Updated version references** - Changed to current versions (Ollama v0.11.x, Continue v1.1.x)
✅ **Added link to recommended models** - Linked to `/customization/models#recommended-models` for additional guidance

Thanks for the thorough review! The documentation should now be more helpful for users setting up Ollama with Continue.

bdougie commented:
> You still need to select the Autocomplete model manually. Same with Rerank. Autodetect wont select it for you, but only selects the chat model from the dropdown.

Sorry that wasn't a useful response. To be clear in the config.yaml, you will need to make sure autocomplete is explicitly listed. I am actually updating the guide now with this context.  

```yml
  - name: Autodetect
    provider: ollama
    model: AUTODETECT
    roles:
      - chat
      - edit
      - apply
      - rerank
      - autocomplete
```

sestinj commented:
:tada: This PR is included in version 1.10.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@continuedev/openai-adapters/v/1.10.0)
- [GitHub release](https://github.com/continuedev/continue/releases/tag/%40continuedev/openai-adapters%401.10.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
